generators/entity/templates/server/src/main/java/package/service/mapper/_BaseEntityMapper.java

```diff
@@ -0,0 +1,36 @@
+<%#
+ Copyright 2013-2017 the original author or authors.
+
+ This file is part of the JHipster project, see https://jhipster.github.io/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.service.mapper;
+
+import java.util.List;
+
+/**
+ * Contract for a generic entity to dto mapper.
+ */
+
+public interface EntityMapper <D, E> {
```

Comment:
I fully agree with the intention of keeping the type names readable/descriptive. 

I followed the Sun/Oracle [convention](http://stackoverflow.com/questions/2900881/generic-type-parameter-naming-convention-for-java-with-multiple-chars) which recommends "single, uppercase letters for type parameters". This helps developers effortlessly distinguish the difference between a type variable and an ordinary class or interface name. Also, this convention guarantees consistency. 

Another option I came across is adding details in the javadoc comments section. This approach follows the convention (thereby removing the possible confusion) and adds the detail you wanted. 

```
/**
 Contract for a generic dto to entity mapper.
 @param <D> - DTO
 @param <E> - Entity
 */
public interface EntityMapper <D, E> {
```

If it doesn't look good, I can change it as per your suggestion.